PatternPal/PatternPal.LoggingServer/Services/LoggerService.cs

```diff
@@ -85,90 +112,87 @@ public override async Task<LogResponse> Log(LogRequest request, ServerCallContex
         /// </summary>
         /// <param name="compressed">A bytestring representing the compressed data</param>
         /// <returns>The CodeState-GUID</returns>
-        /// <exception cref="RpcException">Is thrown when the supplied bytestring was not a valid zip archive.</exception>
-        private Guid ParseCodeState(byte[] compressed)
+        private Guid? ParseCodeState(byte[] compressed)
         {
-            Guid codeStateId = Guid.NewGuid();
-
-            if (!IsZipArchive(compressed))
+            try
             {
-                throw new RpcException(new Status(StatusCode.InvalidArgument, "Data is not a valid zip archive"));
-            }
+                Guid codeStateId = Guid.NewGuid();
 
-            string codeStatePath = Path.Combine(_codeStateDirectory, codeStateId.ToString());
-            Directory.CreateDirectory(codeStatePath);
+                string codeStatePath = Path.Combine(_codeStateDirectory, codeStateId.ToString());
 
-            // Convert the byte array to a zip file and extract it
-            using MemoryStream ms = new(compressed);
-            using ZipArchive archive = new(ms);
-            foreach (ZipArchiveEntry entry in archive.Entries)
-            {
-                string fullPath = Path.Combine(codeStatePath, entry.FullName);
-                string? directory = Path.GetDirectoryName(fullPath);
-                // Create the directory if it does not exist (for nested directories)
-                if (directory != null)
-                {
-                    Directory.CreateDirectory(directory);
-                }
+                // Convert the byte array to a zip file and extract it
+                using MemoryStream ms = new(compressed);
+                using ZipArchive archive = new(ms);
 
-                // Skip non-cs files
-                if (!entry.FullName.EndsWith(".cs"))
+                Directory.CreateDirectory(codeStatePath);
+                foreach (ZipArchiveEntry entry in archive.Entries)
                 {
-                    continue;
+                    string fullPath = Path.Combine(codeStatePath, entry.FullName);
+                    string? directory = Path.GetDirectoryName(fullPath);
+                    // Create the directory if it does not exist (for nested directories)
+                    if (directory != null)
+                    {
+                        Directory.CreateDirectory(directory);
+                    }
+
+                    // Skip non-cs files
+                    if (!entry.FullName.EndsWith(".cs"))
+                    {
+                        continue;
+                    }
+
+                    entry.ExtractToFile(fullPath, true);
                 }
 
-                entry.ExtractToFile(fullPath, true);
+                return codeStateId;
+            }
+            catch (InvalidDataException e)
+            {
+                _logger.LogError(e, "Error while parsing code state");
```

Comment:
Do we need to separately catch the two if we are handling them the same way anyways?